src/provider/intents.ts

```diff
@@ -0,0 +1,32 @@
+import {Context, ContactContext} from '../client/context';
+import {Intents} from '../client/intents';
+
+/**
+ * TypeScript definitions for FDC3-defined intents.
+ *
+ * These structures are defined by the Intents FDC3 working group. The definitions here are based on current
+ * proposals and are not final. There may be minor differences between the current spec and the definitions here,
+ * in order to support this demo.
+ */
+
+/**
+ * General-purpose intent type
+ */
+export interface Intent<T=string, C extends Context=Context> {
+    type: T;
+    context: C;
+    target?: string;
+}
+
+// Built-in intents
+export type DialCallIntent = Intent<typeof Intents.DIAL_CALL, ContactContext>;
```

Comment:
Don't think you need the `typeof` here or the next line.

Reply:
Maybe but this is just a cut and paste from elsewhere, so I am not sure I want to change it.